instruqt/delivering-as-an-appliance/01-specifying-the-cluster/assignment.md

```diff
@@ -1,26 +1,223 @@
+
+```
+helm package -u slackernews -d release
+```
+
+We're then going to add a file that lets the Admin Console know about the Helm
+chart. It uses this file to identify which chart to install and pass the
+appropriate values. We'll look at the passing values in a later section of the
+lab when we set up the configuration screen for our application.
+
+Go to the "Release Editor" tab to add a file to your release.
+
+![Creating a manifest file describing your Helm chart](../assets/creating-the-helmchart-object.png)
+
+The editor may not open your new file automatically. If it doesn't, click on it
+to open it. Add the following content to the file. Note that it looks like a
+Kubernetes custom resource, but it's really not. Instead, it's processed by the
+Admin Console to avoid the complexity of creating a CRD and the relevant
+controllers.
+
+```
+apiVersion: kots.io/v1beta2
+kind: HelmChart
+metadata:
+  name: slackernews
+spec:
+  # chart identifies a matching chart from a .tgz
+  chart:
+    name: slackernews
+    chartVersion: 0.6.0
+
+  # values are used in the customer environment, as a pre-render step
+  # these values will be supplied to helm template
+  values: {}
+```
+
+The name and version in this file need to match the metadata for our
+Slackernews Helm chart to identify it correctly. You're not limited to only one
+Helm chart as part of your application. Including multiple `HelmChart` objects
+let's the Admin Console know it has to install multiple Helm charts. For
+Slackernews, we have only a single chart.
+
+### Including the Embedded Cluster
+
+We showed a simple Embedded Cluster configuration earlier in the lab. Let's use
+that basic configuration for Slackernews. Create another file in the `release`
+folder named `embedded-cluster.yaml` and copy the contents of the configuration
+into it.
+
+```
+apiVersion: embeddedcluster.replicated.com/v1beta1
+kind: Config
+spec:
+  version: [[ Instruqt-Var key="EMBEDDED_CLUSTER_VERSION" hostname="shell" ]]
+```
+
+Releasing the Appliance
+========================
+
+With the three files we have, we can now release Slackernwes Kubernetes
+appliance. We're going to use the `replicated` command-line tool for to create
+the release. The first thing we'll do is set some environment variables to let
+the command know who we are what application we're working with. Use the
+"Shell" tab to set these variables:
+
+```
+export REPLICATED_API_TOKEN="[[ Instruqt-Var key="REPLICATED_API_TOKEN" hostname="shell" ]]"
+export REPLICATED_APP="[[ Instruqt-Var key="REPLICATED_APP" hostname="shell" ]]"
+```
+
+We create a new release with the `release create` subcommand for `replicated`.
+If you've completed some of our other labs you may notice a difference here,
+we're using a flag `--yaml-dir` instead of `--chart`. That's because this
+release requires the additional configuration files we've created.
+
+```
+replicated release create --yaml-dir ./release
```

Comment:
Do we not lint because of the bad linter warning like around Helm status informers? Or just not necessary perhaps.

Reply:
I did notice it wasn't linting when I ran through today, and I'm not certain whether lint is good or not without fine tuning the rules (which we can't easily do here).

Reply:
I'm going to leave linting out for now to avoid having to explain the warnings.